### ISSUE TITLE
Option to remove Modal Header

### ISSUE THREAD

**Issue by HardeepAsrani** [WordPress/gutenberg#13207]
As of now, Modal component doesn't allow you to disable its header, in case if you wanted to build your own header to match with your details, something I'm trying to do.

This limits the flexibility of the component, and there should be an option to disable the header of the modal.

We can right now disable the close button and leave the title empty, but then it just leaves an empty `div` tag which isn't ideal.

I've prepared a PR for this and can send if this is something people consider to be useful.

**Comment by draganescu** [comment]
Howdy  @HardeepAsrani  in the light of #13208 being closed should this remain open?

**Comment by HardeepAsrani** [comment]
Closed @draganescu 😄
